internal/integrations/github/github_test.go

```diff
@@ -0,0 +1,78 @@
+package github
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/google/go-github/v26/github"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type MockChecksService struct {
+	mock.Mock
+}
+
+func (m *MockChecksService) ListCheckRunsForRef(context context.Context, owner string, repo string, ref string, opt *github.ListCheckRunsOptions) (*github.ListCheckRunsResults, *github.Response, error) {
+	args := m.Called()
+	results := args.Get(0)
+
+	if results != nil {
+		return args.Get(0).(*github.ListCheckRunsResults), nil, nil
+	}
+
+	return nil, nil, args.Error(2)
+
+}
+
+func TestGetTravisCIBuildURLForRef_Success(t *testing.T) {
+	ctx := context.Background()
+	g := New(ctx, "Wattpad", "fake-access-token")
+
+	m := new(MockChecksService)
+	m.On("ListCheckRunsForRef").Return(&github.ListCheckRunsResults{
+		CheckRuns: []*github.CheckRun{
+			&github.CheckRun{
+				DetailsURL: github.String("https://travis-ci.com/Wattpad/highlander/builds/115827260"),
+				App: &github.App{
+					Name: github.String("Travis CI"),
+				},
+			},
+		},
+	}, nil, nil)
+	g.Checks = m
+
+	url, _ := g.GetTravisCIBuildURLForRef(ctx, "highlander", "master")
+
+	assert.Equal(t, "https://travis-ci.com/Wattpad/highlander/builds/115827260", url)
+}
+
+func TestGetTravisCIBuildURLForRef_Empty(t *testing.T) {
+	ctx := context.Background()
+	g := New(ctx, "Wattpad", "fake-access-token")
+
+	m := new(MockChecksService)
+	m.On("ListCheckRunsForRef").Return(&github.ListCheckRunsResults{
+		CheckRuns: []*github.CheckRun{},
+	}, nil, nil)
+	g.Checks = m
+
+	_, err := g.GetTravisCIBuildURLForRef(ctx, "highlander", "master")
+
+	assert.EqualError(t, errTravisCIBuildNotFound, err.Error())
+}
+
+func TestGetTravisCIBuildURLForRef_Error(t *testing.T) {
+	ctx := context.Background()
+	g := New(ctx, "Wattpad", "fake-access-token")
+
+	fakeError := errors.New("fake")
+	m := new(MockChecksService)
+	m.On("ListCheckRunsForRef").Return(nil, nil, fakeError)
+	g.Checks = m
+
+	_, err := g.GetTravisCIBuildURLForRef(ctx, "highlander", "master")
+
+	assert.EqualError(t, fakeError.Error(), err)
```

Comment:
It looks like the behaviour we want to test here is that when the `ChecksService` returns an error, then `GetTravisCIBuildURLForRef` should return an error. However, this test case would break if `GetTravisCIBuildURLForRef` added context to the checks error with `errors.Wrap`, even though that's a valid way to implement the desired behaviour.

Since the error value is arbitrary (it's defined by the test case), I would prefer just `assert.Error(t, err)`. If it's particularly important that the `GetTravisCIBuildURLForRef` error is related to the original checks error, then we could do `assert.Equal(t, fakeError, errors.Cause(err))`.